pulser/simulation/simulation.py

```diff
@@ -230,19 +277,59 @@ def adapt(full_array: np.ndarray) -> np.ndarray:
         def make_vdw_term() -> float:
             """Construct the Van der Waals interaction Term.
 
-            For each pair of qubits, calculate the distance between them, then
-            assign the local operator "sigma_rr" at each pair. The units are
-            given so that the coefficient includes a 1/hbar factor.
+            For each pair of qubits, calculate the distance between them,
+            then assign the local operator "sigma_rr" at each pair.
+            The units are given so that the coefficient includes a
+            1/hbar factor.
             """
             vdw = 0
             # Get every pair without duplicates
             for q1, q2 in itertools.combinations(self._qdict.keys(), r=2):
                 dist = np.linalg.norm(
                     self._qdict[q1] - self._qdict[q2])
                 U = 0.5 * self._seq._device.interaction_coeff / dist**6
-                vdw += U * self._build_operator('sigma_rr', q1, q2)
+                vdw += U * self.build_operator([('sigma_rr', [q1, q2])])
             return vdw
 
+        def make_xy_term() -> float:
+            """Construct the XY interaction Term.
+
+            For each pair of qubits, calculate the distance between them,
+            then assign the local operator "sigma_du * sigma_ud" at each pair.
+            The units are given so that the coefficient
+            includes a 1/hbar factor.
+            """
+            xy = 0
+            # Get every pair without duplicates
+            for q1, q2 in itertools.combinations(self._qdict.keys(), r=2):
+                dist = np.linalg.norm(
+                    self._qdict[q1] - self._qdict[q2])
+                mag_norm = np.linalg.norm(self._seq.magnetic_field[0:2])
+                if mag_norm < 1e-8:
+                    cosine = 0.
+                else:
+                    cosine = np.dot(
+                        (self._qdict[q1] - self._qdict[q2]),
+                        self._seq.magnetic_field[0:2]
+                        ) / (dist * mag_norm)
+                U = 0.5 * self._seq._device.interaction_coeff_xy * \
+                    (1 - 3 * cosine ** 2) / dist**3
+                xy += U * self.build_operator(
+                    [('sigma_du', [q1]),
+                     ('sigma_ud', [q2])]
+                )
+            return xy
+
+        def make_interaction_term() -> float:
+            """Construct interaction term.
+
+            Depending on the type of interaction.
+            """
+            if self._interaction == 'ising':
+                return make_vdw_term()
+            else:
+                return make_xy_term()
```

Comment:
You can actually do this synthetically with:
```suggestion
        make_interaction_term = make_xy_term if self._interaction == 'XY' else make_vdw_term
```